nvdaHelper/remote/winword.cpp

```diff
@@ -690,6 +694,272 @@ inline bool generateFootnoteEndnoteXML(IDispatch* pDispatchRange, wostringstream
+	// The following case should handle where we have the page break character ('0x0c') shown with '|p|'
+	//	first section|p|
+	//	second section.
+	// range.Sections[1].pageSetup.SectionStart tells you how the section started, so we need to know
+	// the next sections start type to report what kind of break this is. To do this we need to expand
+	// the range, get the section start type, remove the page break character, and insert an attribute
+	// for the break type.
+	IDispatchPtr pDispatchRangeDup = nullptr;
+	auto res = _com_dispatch_raw_propget( pDispatchRange, wdDISPID_RANGE_DUPLICATE, VT_DISPATCH, &pDispatchRangeDup);
+	if( res != S_OK || !pDispatchRangeDup ) {
+		LOG_DEBUGWARNING(L"error duplicating the range.");
+		return {};
+	}
+
+	// we assume that we are 1 character away from the next section, this should be the value for PAGE_BREAK_VALUE ("0x0c")
+	const int unitsToMove = 1;
+	int unitsMoved=-1;
+	res = _com_dispatch_raw_method(pDispatchRangeDup,wdDISPID_RANGE_MOVEEND,DISPATCH_METHOD,VT_I4,&unitsMoved,L"\x0003\x0003",wdCharacter,unitsToMove);
+	if( res !=S_OK || unitsMoved<=0 || !pDispatchRangeDup) {
+		LOG_DEBUGWARNING(L"error moving the end of the range");
+		return {};
+	}
+
+	IDispatchPtr pDispatchSections = nullptr;
+	res = _com_dispatch_raw_propget( pDispatchRangeDup, wdDISPID_RANGE_SECTIONS, VT_DISPATCH, &pDispatchSections);
+	if( res != S_OK || !pDispatchSections ) {
+		LOG_DEBUGWARNING(L"error getting sections from range");
+		return {};
+	}
+
+	int count = -1;
+	res = _com_dispatch_raw_propget( pDispatchSections, wdDISPID_SECTIONS_COUNT, VT_I4, &count);
+	if( res != S_OK || count != 2 ) { 
+		LOG_DEBUGWARNING(L"error getting section count. There should be exactly 2 sections, count: " << count);
+		return {};
+	}
+
+	// we make the assumption that the second section will always be the one we want. We also assume that the section
+	// count was 1 before expanding the range.
+	const int sectionToGet = 2;
+	IDispatchPtr pDispatchItem = nullptr;
+	res = _com_dispatch_raw_method( pDispatchSections, wdDISPID_SECTIONS_ITEM, DISPATCH_METHOD, VT_DISPATCH, &pDispatchItem, L"\x0003", sectionToGet);
+	if( res != S_OK || !pDispatchItem){
+		LOG_DEBUGWARNING(L"error getting section item");
+		return {};
+	}
+
+	IDispatchPtr pDispatchPageSetup = nullptr;
+	res = _com_dispatch_raw_propget( pDispatchItem, wdDISPID_SECTION_PAGESETUP,  VT_DISPATCH, &pDispatchPageSetup);
+	if( res != S_OK || !pDispatchPageSetup){
+		LOG_DEBUGWARNING(L"error getting pageSetup");
+		return {};
+	}
+
+	int type = -1;
+	res = _com_dispatch_raw_propget( pDispatchPageSetup, wdDISPID_PAGESETUP_SECTIONSTART, VT_I4, &type);
+	if( res != S_OK || type < 0){
+		LOG_DEBUGWARNING(L"error getting section start");
+		return {};
+	}
+
+	LOG_DEBUGWARNING(L"Got Type: " << type);
+	return type;
+}
+
+std::experimental::optional<int>
+getStartOfRangeDistanceFromStartOfDocument(IDispatchPtr pDispatchRange) {

```

Comment:
the name "startOfDocument" bothers me, as  it could mean in terms of offsets. Perhaps "edgeOfDocument" may be better?